### PR TITLE
Binary search C++ added

### PR DESCRIPTION
**Pull request by shauryauppal**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #{ISSUE_NUMBER}

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [ ] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [ ] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [ ] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

**Comment by shauryauppal** [comment]
#54

**Comment by shauryauppal** [comment]
@sangamcse review this also.

**Comment by shauryauppal** [comment]
@sangamcse do I need to make these changes?

**Comment by sangamcse** [comment]
@shauryauppal Yes. Also, rebase your local repo and then do the changes.
Follow https://github.com/NITSkmOS/Algorithms/blob/master/GIT_BASICS.md this guide if you are new to git

File: binary_search/C++/binarysearch_STL.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#include <bits/stdc++.h>
```
Comment (sangamcse) [inline binary_search/C++/binarysearch_STL.cpp]: This file is not for this issue. So remove it.

File: binary_search/C++/binarysearch.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#include <bits/stdc++.h>
+using namespace std;
+bool binarysearch(const int A[],int n, int ele)
```
Comment (sangamcse) [inline binary_search/C++/binarysearch.cpp]: code comments for this method

**Comment by sangamcse** [comment]
Closing due to inactivity.

**Pull request closed without merging**